core/src/main/scala/org/apache/spark/sql/store/StoreUtils.scala

```diff
@@ -400,16 +400,11 @@ object StoreUtils {
         } else if (v == NONE) {
           EMPTY_STRING
         } else {
-          s"$GEM_EVICTION_BY $v "
+          s"$GEM_EVICTION_BY $v $GEM_OVERFLOW "
         }
       }).getOrElse(defaultEviction))
     }
 
-    if (hasOverflow) {
-      parameters.remove(OVERFLOW)
-      sb.append(s"$GEM_OVERFLOW ")
-    }
-
```

Comment:
What if user explicitly sets overflow option as false? This change will not honor that option. I think the better way will be to set hasOverflow getOrElse() block to be always true. Then note when the EVICTION_BY=NONE case (that appends EMPTY_STRING) with some boolean flag, then add the GEM_OVERFLOW in this block if its not the EVICTION_BY=NONE case.